planner/memo/expr_iterator.go

```diff
@@ -71,6 +72,10 @@ func (iter *ExprIter) Next() (found bool) {
 			return false
 		}
 
+		if len(iter.Children) == 0 {
```

Comment:
This change is not related to this PR. Here is the example to explain this change.
Suppose we have such a pattern: `Proj -> Sel`. (Note there is no Any)
And we have a memo like:
```
Group#0:
    Projection_0 Input: [Group#1]
Group#1:
    Selection_1 Input: [...]
    Selection_2 Input: [...]
```
Before this change, we will only match the pattern once(`Projection_0->Selection_1`). But actually, we want to match it twice(`Projection_0->Selection_1`, `Projection_0->Selection_2`).

Reply:
By the way, the unit tests added by this PR require this change.

Reply:
BTW, the unit tests added by this PR require this change.

Reply:
Could you add this to the comments?

Reply:
Which test do you mean? I removed it and everything still passes.

Reply:
Not only here, there are 3 changes like `if len(iter.Children) == 0`. 
I removeed that test before, because I think it is not useful. But I just add a new test case for this in the last commit, you can check it.